panic in rpc.Conn.sendCap

I'm getting this stack trace having attempted to port go.sandstorm to the v3 branch:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x6935dc]

goroutine 51 [running]:
zombiezen.com/go/capnproto2/rpc.(*Conn).sendCap(0xc4200c4280, 0xc42010ec30, 0x800000050, 0x1, 0xfffffffffffffffe, 0x1, 0xc42008ea40, 0x70c2e0, 0xc42007cde0, 0xffffffffffffff00, ...)
	/home/isd/go/src/zombiezen.com/go/capnproto2/rpc/export.go:99 +0xec
zombiezen.com/go/capnproto2/rpc.(*Conn).fillPayloadCapTable(0xc4200c4280, 0xc42010ec30, 0x30, 0x2, 0xffffffffffffffff, 0x0, 0xc42008a080, 0x1, 0x1, 0xc4200f4da0, ...)
	/home/isd/go/src/zombiezen.com/go/capnproto2/rpc/export.go:137 +0x2b1
zombiezen.com/go/capnproto2/rpc.(*answer).sendReturn(0xc420082ea0, 0xc4200f4da0, 0x1, 0x1, 0xc4200f4da0, 0x1, 0x1, 0x0, 0xc42010ec30)
	/home/isd/go/src/zombiezen.com/go/capnproto2/rpc/answer.go:211 +0x11b
zombiezen.com/go/capnproto2/rpc.(*answer).Return(0xc420082ea0, 0x0, 0x0)
	/home/isd/go/src/zombiezen.com/go/capnproto2/rpc/answer.go:176 +0x2a8
zombiezen.com/go/capnproto2/server.(*Server).start.func1(0xc42012a118, 0x7c4fc0, 0xc42008e940, 0xc420128580, 0xdbb4d798ea67e2e7, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
	/home/isd/go/src/zombiezen.com/go/capnproto2/server/server.go:268 +0x271
created by zombiezen.com/go/capnproto2/server.(*Server).start
	/home/isd/go/src/zombiezen.com/go/capnproto2/server/server.go:263 +0x5fe
```

It looks like sendCap isn't accounting for the possibility that an entry in the exports table could be nil, but I don't understand what's going on well enough to know what the right fix is.

--- a/rpc/export.go
+++ b/rpc/export.go
@@ -96,6 +96,9 @@ func (c *Conn) sendCap(d rpccp.CapDescriptor, client *capnp.Client, state capnp.
 
 	// Default to sender-hosted (export).
 	for id, ent := range c.exports {
+		if ent == nil {
+			continue
+		}
 		if ent.client.IsSame(client) {
 			ent.wireRefs++
 			d.SetSenderHosted(uint32(id))

